cirq-core/cirq/ops/pauli_string.py

```diff
@@ -981,31 +980,28 @@ def conjugated_by(self, clifford: 'cirq.OP_TREE') -> 'PauliString':
         ps = PauliString(qubit_pauli_map=self._qubit_pauli_map, coefficient=self.coefficient)
         all_ops = list(op_tree.flatten_to_ops(clifford))
         all_qubits = set.union(set(self.qubits), [q for op in all_ops for q in op.qubits])
-
         # Iteratively calculate the conjugation in reverse order of ops.
         for op in all_ops[::-1]:
             # To calcuate the conjugation of P (`ps`) with respect to C (`op`)
             # Decompose P = Pc⊗R, where Pc acts on the same qubits as C, R acts on the remaining.
             # Then the conjugation = (C^{-1}⊗I·Pc⊗R·C⊗I) = (C^{-1}·Pc·C)⊗R.
 
             # Isolate R
-            remain: 'cirq.PauliString' = PauliString()
-            for q in all_qubits:
-                pauli = ps.get(q)
-                if pauli is not None and not q in op.qubits:
-                    remain *= pauli(q)
+            remain: 'cirq.PauliString' = PauliString(
+                *(pauli(q) for q in all_qubits - set(op.qubits) if (pauli := ps.get(q)) is not None)
+            )
 
             # Initialize the conjugation of Pc.
             conjugated: 'cirq.DensePauliString' = (
                 dense_pauli_string.DensePauliString(pauli_mask=[identity.I for _ in op.qubits])
-                * self.coefficient
+                * ps.coefficient
```

Comment:
This is a bug of the previous pr I submitted. We should roll against ps in the iteration, coeff should be from ps not self, found this by enforcing unitary checks in tests for all conjugated_by tests, previously if expected is given, unitary isn't checked. Fix it here.